inference-engine/src/transformations/src/transformations/common_optimizations/random_uniform_fusion.cpp

```diff
@@ -0,0 +1,161 @@
+        const auto& pattern_map = m.get_pattern_value_map();
+        const auto data = pattern_map.at(data_pattern);
+        const auto random_uniform = pattern_map.at(random_uniform_pattern);
+        const auto mul_constant = pattern_map.at(mul_const_pattern);
+        const auto ru = std::dynamic_pointer_cast<opset8::RandomUniform>(random_uniform.get_node_shared_ptr());
+        if (!ru)
+            return false;
+
+        const auto mul_const = std::dynamic_pointer_cast<opset8::Constant>(mul_constant.get_node_shared_ptr());
+        if (!mul_const)
+            return false;
+
+        auto const_shape = mul_const->get_shape();
+        size_t const_shape_size = shape_size(const_shape);
+
+        if (const_shape_size != 1)
+            return false;
+        const auto value = mul_const->cast_vector<double>();
+        auto new_mul_const = op::Constant::create(ru->get_out_type(), Shape{}, value);
+
+        if (pattern_map.count(convert_pattern)) {
+            const auto& mul = pattern_map.at(mul_pattern);
+            const auto ml = std::dynamic_pointer_cast<opset8::Multiply>(mul.get_node_shared_ptr());
+            if (!ml)
+                return false;
+
+            const auto& convert = pattern_map.at(convert_pattern);
+            const auto cvt = std::dynamic_pointer_cast<opset8::Convert>(convert.get_node_shared_ptr());
+            if (!cvt)
+                return false;
+            const auto new_mul1 = ml->clone_with_new_inputs({ru->input_value(1), new_mul_const});
+            const auto new_mul2 = ml->clone_with_new_inputs({ru->input_value(2), new_mul_const});
+            const auto new_ru = ru->clone_with_new_inputs({data, new_mul1, new_mul2});
+            new_ru->set_friendly_name(m.get_match_root()->get_friendly_name());
+            const auto new_ru_conv = cvt->clone_with_new_inputs({new_ru});
+            copy_runtime_info({ru, cvt}, {new_ru, new_ru_conv});
+            ngraph::replace_node(m.get_match_root(), new_ru_conv);
+
+        } else {
+            const auto& mul = pattern_map.at(mul_pattern);
+            const auto ml = std::dynamic_pointer_cast<opset8::Multiply>(mul.get_node_shared_ptr());
+            if (!ml)
+                return false;
+            const auto new_mul1 = ml->clone_with_new_inputs({ru->input_value(1), new_mul_const});
+            const auto new_mul2 = ml->clone_with_new_inputs({ru->input_value(2), new_mul_const});
+            const auto new_ru = ru->clone_with_new_inputs({data, new_mul1, new_mul2});
+            new_ru->set_friendly_name(m.get_match_root()->get_friendly_name());
+            copy_runtime_info({mul.get_node_shared_ptr(), ru}, {new_mul1, new_mul2, new_ru});
+            ngraph::replace_node(m.get_match_root(), new_ru);
+        }
+
+        return true;
+    };
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(mul_pattern, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+ngraph::pass::RandomUniformMinValFusion::RandomUniformMinValFusion() {
```

Comment:
please rename transformation to `RandomUniformAddFusion`

Reply:
Done.